engine/src/cmdsm.cpp

```diff
@@ -1327,6 +1327,14 @@ void MCArrayOp::exec_ctxt(MCExecContext &ctxt)
 	}
 	else
 	{
+		/* Delimiters cannot be empty for split */
+		if (MCStringIsEmpty(*t_element_del) ||
```

Comment:
This check needs to go into the ExecSplit* methods. This is a runtime check after type-conversion - the code in the exec_ctxt methods should be (as far as is possible) type-convert, select method, dispatch.